Sources/Algorithms/SlidingWindows.swift

```diff
@@ -84,9 +84,198 @@ extension SlidingWindows: Collection {
     )
   }
   
-  // TODO: Implement distance(from:to:), index(_:offsetBy:) and
-  // index(_:offsetBy:limitedBy:)
+  public func index(_ i: Index, offsetBy distance: Int) -> Index {
+    guard distance != 0 else { return i }
+    
+    return distance > 0
+      ? offsetForward(i, by: distance)
+      : offsetBackward(i, by: -distance)
+  }
+  
+  public func index(
+    _ i: Index,
+    offsetBy distance: Int,
+    limitedBy limit: Index
+  ) -> Index? {
+    guard distance != 0 else { return i }
+    guard limit != i else { return nil }
+    
+    if distance > 0 {
+      return limit > i
+        ? offsetForward(i, by: distance, limitedBy: limit)
+        : offsetForward(i, by: distance)
+    } else {
+      return limit < i
+        ? offsetBackward(i, by: -distance, limitedBy: limit)
+        : offsetBackward(i, by: -distance)
+    }
+  }
+  
+  private func offsetForward(_ i: Index, by distance: Int) -> Index {
+    offsetForward(i, by: distance, limitedBy: endIndex)!
```

Comment:
Instead of force unwrapping here (and `offsetBackward`) I would add an error message describing why the offset was not valid. We did something similar [here](https://github.com/apple/swift-algorithms/pull/42/files#diff-2d7e1f0b61f31e10e56c985d329b222f086037d37960895c16669eaedd036515R79)